.github/workflows/ci.yml

```diff
@@ -12,32 +12,53 @@ env:
 
 jobs:
   win64-native:
-    name: "x86_64: Windows, VS 2022"
+    name: ${{ matrix.configuration.job_name }}
     # See: https://github.com/actions/runner-images#available-images.
     runs-on: windows-2022
 
     strategy:
       fail-fast: false
       matrix:
-        lib_type: ['shared', 'static']
+        configuration:
+          - job_name: 'x64 (MSVC): Windows (VS 2022, shared)'
+            cmake_options: '-A x64 -DBUILD_SHARED_LIBS=ON'
+          - job_name: 'x64 (MSVC): Windows (VS 2022, static)'
+            cmake_options: '-A x64 -DBUILD_SHARED_LIBS=OFF'
+          - job_name: 'x64 (MSVC): Windows (VS 2022, int128_struct)'
+            cmake_options: '-A x64 -DSECP256K1_TEST_OVERRIDE_WIDE_MULTIPLY=int128_struct'
+          - job_name: 'x64 (MSVC): Windows (VS 2022, int128_struct with __(u)mulh)'
+            cmake_options: '-A x64 -DSECP256K1_TEST_OVERRIDE_WIDE_MULTIPLY=int128_struct'
+            cpp_flags: '/DSECP256K1_MSVC_MULH_TEST_OVERRIDE'
+          - job_name: 'x86 (MSVC): Windows (VS 2022)'
+            cmake_options: '-A Win32'
 
     steps:
       - name: Checkout
         uses: actions/checkout@v3
 
       - name: Generate buildsystem
-        run: cmake -E env CFLAGS="/WX" cmake -B build -A x64 -DSECP256K1_ENABLE_MODULE_RECOVERY=ON -DSECP256K1_BUILD_EXAMPLES=ON -DBUILD_SHARED_LIBS=${{ matrix.lib_type == 'shared' && 'ON' || 'OFF' }}
+        run: cmake -E env CFLAGS="/WX ${{ matrix.configuration.cpp_flags }}" cmake -B build -DSECP256K1_ENABLE_MODULE_RECOVERY=ON -DSECP256K1_BUILD_EXAMPLES=ON ${{ matrix.configuration.cmake_options }}
 
       - name: Build
         run: cmake --build build --config RelWithDebInfo -- /p:UseMultiToolTask=true /maxCpuCount
 
+      - name: Binaries info
+        # Use the bash shell included with Git for Windows.
+        shell: bash
+        run: |
+          cd build/src/RelWithDebInfo && file *tests.exe bench*.exe libsecp256k1-*.dll || true
+
       - name: Check
         run: |
           ctest -C RelWithDebInfo --test-dir build -j ([int]$env:NUMBER_OF_PROCESSORS + 1)
           build\src\RelWithDebInfo\bench_ecmult.exe
           build\src\RelWithDebInfo\bench_internal.exe
           build\src\RelWithDebInfo\bench.exe
 
+      - uses: ilammy/msvc-dev-cmd@v1
+      - name: C++ (public headers)
+        run: cl.exe -c -WX -TP include/*.h
```

Comment:
Is the `ilammy/msvc-dev-cmd@v1` required only for the C++ compiler? To be honest, we could drop that part then. But do you really need this? Can't we use the already installed MSVC? (I mean, we use it above to compile the library, so it must be somewhere on these machines, right?)

Reply:
We can use `C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.35.32215/bin/HostX64/x64/cl.exe`, but it needs to be updated every time GH Windows image is updated due to the version part in the path.

Reply:
Couldn't we  just invoke it with a glob like `C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/*/bin/HostX64/x64/cl.exe`? (or whatever the windows equivalent of this is, or use bash).

Reply:
> Couldn't we just invoke it with a glob...

I guess not:

```
  dir "C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/"
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"

    Directory: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d----            8/4/2023 10:41 PM                14.16.27023
d----            8/4/2023 10:39 PM                14.29.30133
d----            8/4/2023 11:07 PM                14.35.32215
d----            8/4/2023 10:28 PM                14.36.32532
```

Reply:
> To be honest, we could drop that part then.

It has been dropped.

Reply:
Okay, after some digging, I'm actually convinced that https://github.com/ilammy/msvc-dev-cmd is the commonly accepted way of doing this. I also found https://github.com/actions/runner-images/issues/5986#issuecomment-1201012941, but not sure if it's better.

So I'm leaning towards bringing https://github.com/ilammy/msvc-dev-cmd back. Then it will be even cleaner to do this step before the regular CMake invocations, too? And If this ever breaks in the future, it should still be easy to get rid of it. Sorry for the back and forth.

Reply:
> It seems to have been added already.

It works!

Reply:
> ... it seems that CMake works here more or less by accident.

The Windows image has [CMake installed](https://github.com/actions/runner-images/blob/main/images/win/Windows2022-Readme.md#tools) separately from Visual Studio tools, which is nice.

In turn, CMake itself is smart enough to find paths to all tools it requires.

Only direct invocation of the `cl.exe` compiler requires having it in the `%PATH%`.

That's why I do prefer keeping these lines of code together.

Reply:
Ok, makes sense. Keeping them together also self-documents the fact that the external action is only required for that single line of code.